### PR TITLE
Remove Business Portal, add SSO configuration models

### PR DESCRIPTION
**Pull request by Hinton**
## Objective
Adds the necessary model changes to support setting up and configuring sso using the web vault.

### Noteworthy Code Changes
- **angular/src/components/environment.component.ts**: Remove `enterpriseUrl`, since the service is now gone.
- **common/src/services/environment.service.ts**: Remove enterprise related methods.
- **common/src/abstractions/api.service.ts**: Remove `getEnterprisePortalSignInToken`, add `getOrganizationSso` and `postOrganizationSso`.
- **common/src/enums/permissions.ts**: Remove `Permissions.AccessBusinessPortal`.
- **common/src/models/response/organization/organizationSsoResponse.ts**: Added response model for SSO configuration.
- **common/src/models/request/organization/organizationSsoRequest.ts**: Added request modal for SSO configuration.

### CODE REVIEW

File: common/src/enums/permissions.ts [hunk 1]
```diff
@@ -1,5 +1,4 @@
 export enum Permissions {
-    AccessBusinessPortal = 0,
```
Comment (MGibson1) [inline common/src/enums/permissions.ts:2]: We should leave this in commented out or deprecated as below (my vote is commented out if there are not upgrade concerns)

We can't reuse values in this enum, so it's best to leave reminders to what permission values were and what are available.

File: common/src/models/request/organization/organizationSsoRequest.ts [hunk 2]
```diff
@@ -0,0 +1,4 @@
+export class OrganizationSsoRequest {
+    enabled: boolean = false;
+    data: any;
```
Comment (MGibson1) [inline common/src/models/request/organization/organizationSsoRequest.ts:3]: Is there some specific reason this needs to be `any` type? The type should be well-defined on the c# side, I'd prefer to use typescripts static-typing power whenever possible.

File: common/src/models/response/organization/organizationSsoResponse.ts [hunk 3]
```diff
@@ -0,0 +1,22 @@
+import { BaseResponse } from '../baseResponse';
+
+export class OrganizationSsoResponse extends BaseResponse {
+    enabled: boolean;
+    data: any;
```
Comment (MGibson1) [inline common/src/models/response/organization/organizationSsoResponse.ts:5]: Same `any` concern

**Review by MGibson1 — CHANGES_REQUESTED** [review]
Looks good, I'm just concerned about the data typing in the sso config request/response.

Nested types can be parsed similar to how we're doing it for `cipherResponses` in [`attachmentUploadDataResponse`](https://github.com/bitwarden/jslib/blob/master/common/src/models/response/attachmentUploadDataResponse.ts)

File: common/src/enums/permissions.ts (hunk 1 above)
Comment (Hinton) [reply common/src/enums/permissions.ts:2]: I believe the enum is only used internally in the projects? So it should be possible to change the enum values.

**Review by MGibson1 — APPROVED** [review]

**Pull request merged**